### DISCUSSION TITLE
How can i use like `Pre Build Hooks` ?

### DISCUSSION THREAD

**Discussion by Zhengqbbb** [vuejs/vitepress#1351] [Q&A] [answered]
I really like the `buildEnd` hook so much , which got me done without relying on vitepress data:
- generate sitemap
- regenerate PWA ServiceWorker
<img width="523" alt="image" src="https://user-images.githubusercontent.com/40693636/190847058-22c6c8b3-774a-4451-af82-4a55198fae3d.png">

but I may also need some hooks like `previous build`, to
- fetch github api data
- change some code
- auto generate some markdown page

I'm not sure if this is a feature request for vitepress. But I want to know if it can be achieved with custom vite plugin.
But I tried the vite plugin api without success.

Wondering if there is a better solution or provide a `hello world` demo

**Comment by brc-dd** [comment, marked as answer]
It should be possible using custom vite plugin -- create one with a transform hook, check if id is your file, then fetch data and update it. I don't have a demo right now. But I'll try to create one. It should be straightforward IG.
